### PR TITLE
added different report types

### PR DESCRIPTION
**Pull request by elimelt**
now actually store fk to other report types

### CODE REVIEW

File: server/report/management/commands/clear_reports.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+from report.models import Report
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Deletes all reports"
+
+    def add_arguments(self, parser):
```
Comment (ShawnCollinge) [inline server/report/management/commands/clear_reports.py:13]: Why do we want to delete reports? I think we should archive instead

File: server/report/models.py [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
-from django.contrib.auth.models import User
-
-# Create your models here.
 class Report(models.Model):
     REPORT_TYPE_CHOICES = [
         ('interview', 'Interview'),
         ('question', 'Question'),
-        # Add other types as needed
+        ('member', 'Member'),
```
Comment (ShawnCollinge) [inline server/report/models.py:11]: Why are we passing tuples why not just make it 'member' etc.

File: server/report/views.py [hunk 3]
```diff
@@ -20,204 +22,173 @@ class ReportOwnerPermission(permissions.BasePermission):
...
-        )
+        return Response({"reports": serializer.data}, status=status.HTTP_200_OK)
 
 
 class GetReportByID(APIView):
-    # should change this to admin only @hoang
-    permission_classes = [IsAuthenticated, IsVerified]
+    permission_classes = [IsAuthenticated, IsVerified, IsAdmin]
+
     def get(self, _, report_id):
```
Comment (ShawnCollinge) [inline server/report/views.py:50]: Why can all members view any report?

File: server/report/views.py [hunk 4]
```diff
@@ -20,204 +22,173 @@ class ReportOwnerPermission(permissions.BasePermission):
...
-            {
-                "report": ReportSerializer(report).data
-            },
-            status=status.HTTP_200_OK
+            {"report": ReportSerializer(report).data}, status=status.HTTP_200_OK
         )
 
 
 class CreateReport(APIView):
     permission_classes = [IsAuthenticated, IsVerified]
```
Comment (ShawnCollinge) [inline server/report/views.py:101]: Shouldn't this just be IsVerified?

File: server/report/views.py [hunk 5]
```diff
@@ -20,204 +22,173 @@ class ReportOwnerPermission(permissions.BasePermission):
...
             return Response(
-                {
-                    "error": "Report not found"
-                },
-                status=status.HTTP_404_NOT_FOUND
+                {"error": "Report not found"}, status=status.HTTP_404_NOT_FOUND
             )
 
-        if 'status' not in request.data:
+        if "status" not in request.data:
```
Comment (ShawnCollinge) [inline server/report/views.py:183]: Shouldn't we only allow certain statuses? Why can this be any value?

**Review by ShawnCollinge — COMMENTED** [review]
Where's the tests?

File: server/report/management/commands/clear_reports.py (hunk 1 above)
Comment (elimelt) [reply server/report/management/commands/clear_reports.py:13]: used while testing, probably shouldn't ever be used in prod though

File: server/report/views.py (hunk 3 above)
Comment (elimelt) [reply server/report/views.py:50]: perms are AND

File: server/report/views.py (hunk 4 above)
Comment (elimelt) [reply server/report/views.py:101]: can be

File: server/report/views.py (hunk 5 above)
Comment (elimelt) [reply server/report/views.py:183]: yes, will add validation to serializer

File: server/report/models.py (hunk 2 above)
Comment (elimelt) [reply server/report/models.py:11]: (DB, presentation)

this is the syntax for choices fields iiuc

File: server/report/views.py (hunk 3 above)
Comment (ShawnCollinge) [reply server/report/views.py:50]: Then shouldn't it just be `IsAdmin`?

Comment (ShawnCollinge) [reply server/report/views.py:50]: Oh you already fixed it haha

**Review by ShawnCollinge — APPROVED** [review]
LGTM

**Pull request merged**